images/api.Dockerfile

```diff
@@ -1,9 +1,10 @@
-FROM golang:1.18.2-alpine3.14 AS builder
+FROM --platform=$BUILDPLATFORM golang:1.18.2-alpine3.14 AS builder
 
 WORKDIR /go/src/github.com/tektoncd/hub
 COPY . .
-
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o api-server ./api/cmd/api/...
+ARG TARGETOS
+ARG TARGETARCH
```

Comment:
can we add a default value for these args so that if some one wants to build just the image for linux/amd64 then it should be straight forward. WDYT?
example
```
ARG TARGETOS=linux
ARG TARGETARCH=amd64
```